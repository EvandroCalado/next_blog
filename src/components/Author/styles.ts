import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 3rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 15rem;
    border: 3px solid ${theme.colors.secondary};
    border-radius: 50%;
  `}
`;

export const Container = styled.div`
`;

export const Title = styled.h2`
${({ theme }) => css`
font-size: ${theme.font.sizes.xlarge};
`}
`;

export const Paragraph = styled.p`
`;
