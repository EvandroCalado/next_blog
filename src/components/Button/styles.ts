import styled, { css } from 'styled-components';
import { ButtonProps } from './';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background-color: ${theme.colors[color]};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    border: none;
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    margin: ${theme.spacings.large} 0;
    border-radius: 0.5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 300ms ease-in-out;
    cursor: pointer;

    &:hover {
      filter: brightness(80%);
    }

    &:disabled {
      background-color: ${theme.colors.mediumGray};
      color: ${theme.colors.darkGray};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 3rem;
      margin-left: 1rem;
    }
  `}
`;
