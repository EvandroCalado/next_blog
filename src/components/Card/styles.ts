import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    a {
      transition: all 300ms ease-in-out;
    }

    a:hover {
      opacity: 1;
      filter: brightness(170%);
    }
`;

export const Cover = styled.div`
  ${({ theme }) => css`
    img {
      width: 100%;
      display: block;
      aspect-ratio: 16/9;
      border-radius: .5rem;
      margin-bottom: ${theme.spacings.small};

    }
  `}
`;

export const Heading = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.smallMedium};
    color: ${theme.colors.darkGray};
    margin-bottom: ${theme.spacings.xsmall};
  `}
`;

export const Paragraph = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.small};
    color: ${theme.colors.primary};
  `}
`;
