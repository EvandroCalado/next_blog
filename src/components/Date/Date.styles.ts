import styled, { css } from 'styled-components';

export const Date = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.secondary};
    font-style: italic;
    margin-right: 1rem;
  `}
`;
