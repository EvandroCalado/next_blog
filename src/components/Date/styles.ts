import styled, { css } from 'styled-components';

export const Date = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.secondary};
    font-style: italic;
    margin-right: 1rem;
  `}
`;
