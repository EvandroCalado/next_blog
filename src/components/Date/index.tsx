import { formatDate } from '../../utils/formatDate';
import * as Styled from './styles';

export type DateProps = {
  date: string;
};

const Date = ({ date }: DateProps) => {
  return <Styled.Date>{formatDate(date)}</Styled.Date>;
};

export default Date;
