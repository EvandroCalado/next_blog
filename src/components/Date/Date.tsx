import { formatDate } from '../../utils/formatDate';
import * as Styled from './Date.styles';

export type DateProps = {
  date: string;
};

const Date = ({ date }: DateProps) => {
  return <Styled.Date>{formatDate(date)}</Styled.Date>;
};

export default Date;
