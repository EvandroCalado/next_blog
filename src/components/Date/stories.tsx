import Date, { DateProps } from '.';
import { Meta, StoryFn } from '@storybook/react';

export default {
  title: 'Date',
  component: Date,
  args: {
    date: '2023-04-25T12:45:35.146Z',
  },
} as Meta;

export const Template: StoryFn<DateProps> = (args) => (
  <div>
    <Date {...args} />
  </div>
);
