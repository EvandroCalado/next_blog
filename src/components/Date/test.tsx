import Date from './';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

const mock = {
  date: '2023-04-25T12:45:35.146Z',
};

describe('<Date />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Date date={mock.date} />);

    expect(screen.getByText('25/04/2023')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
