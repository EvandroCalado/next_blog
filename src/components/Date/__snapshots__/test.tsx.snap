// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Date /> should render 1`] = `
.c0 {
  color: #f24b4b;
  font-style: italic;
  margin-right: 1rem;
}

<div>
  <span
    class="c0"
  >
    25/04/2023
  </span>
</div>
`;
